### ISSUE TITLE
1.2 Update

### ISSUE THREAD

**Issue by NathanKell** [KSP-RO/RealismOverhaul#1436]
- [x] RSS (req Kopernicus)
- [x] FAR https://github.com/ferram4/Ferram-Aerospace-Research/tree/KSP_update (tested, not yet live)
- [x] RF https://github.com/NathanKell/ModularFuelSystem/releases
- [x] TACLS https://github.com/KSP-RO/TacLifeSupport/releases
- [x] RealHeat needs recompile? (https://github.com/KSP-RO/RealHeat/pull/2)
- [x] AJE https://github.com/KSP-RO/AJE/tree/KSP_1.2 (not released, possibly close)
- [x] CLS https://github.com/codepoetpbowden/ConnectedLivingSpace/releases (I haven't tested)
- [x] CBK
- [x] DRE https://github.com/Starwaster/DeadlyReentry/releases (I haven't tested)
- [x] KJR https://github.com/ferram4/Kerbal-Joint-Reinforcement/releases/tag/v3.3.1
- [x] Persistent Rotation http://spacedock.info/mod/447/PersistentRotation
- [x] PF https://github.com/e-dog/ProceduralFairings/releases
- [x] PP https://github.com/Swamp-Ig/ProceduralParts/releases (worked fine w/o RF in KSP 1.2.2)
- [x] RealChute https://github.com/StupidChris/RealChute/releases/tag/v1.4.1.2
- [x] RT https://github.com/RemoteTechnologiesGroup/RemoteTech/releases/tag/1.8.3
- [x] TestFlight https://github.com/KSP-RO/TestFlight/releases/tag/1.8.0.0

**Comment by jwvanderbeck** [comment]
Is anything needed of me for TF or do you guys got it?

**Comment by PhineasFreak** [comment]
Every RO part that utilizes a TACLS converter will need to be updated to the new "TacGenericConverter".

Before:

```
MODULE
{
    name = TacGenericConverter
    converterName = CO2 Scrubber
    conversionRate = 4.0
    inputResources = CarbonDioxide, 0.0058912100, ElectricCharge, 0.010, LithiumHydroxide, 0.0000085683
    outputResources = Water, 0.0032924498, true, Waste, 0.0000257297, false
}
```

After:

```
MODULE
{
    name = TacGenericConverter
    converterName = CO2 Scrubber
    StartActionName = Start CO2 Scrubber
    StopActionName = Stop CO2 Scrubber
    tag = Life Support
    GeneratesHeat = False
    UseSpecialistBonus = True
    SpecialistEfficiencyFactor = 0.2
    SpecialistBonusBase = 0.05
    ExperienceEffect = ConverterSkill
    EfficiencyBonus = 1
    conversionRate = 4.0

    INPUT_RESOURCE
    {
        ResourceName = CarbonDioxide
        Ratio = 0.00589121
    }

    INPUT_RESOURCE
    {
        ResourceName = ElectricCharge
        Ratio = 0.01
    }

    INPUT_RESOURCE
    {
        ResourceName = LithiumHydroxide
        Ratio = 0.0000085683
    }

    OUTPUT_RESOURCE
    {
        ResourceName = Water
        Ratio = 0.0032924498
        DumpExcess = True
    }

    OUTPUT_RESOURCE
    {
        ResourceName = Waste
        Ratio = 0.0000257297
        DumpExcess = False
    }
}
```

**Comment by stratochief66** [comment]
@jwvanderbeck I don't think we know right now. Probably not, but we'll now for sure once we've got RSS & RO rolling again :)

**Comment by blowfishpro** [comment]
I think all the important changes have been made in AJE and RF.  @NathanKell I know you mentioned some improvements that could be made to the ullage code with KSP 1.2's resource flow changes, and I haven't looked at that.

**Comment by lamont-granquist** [comment]
RT has a KSP 1.2.1-compatible version available:  https://github.com/RemoteTechnologiesGroup/RemoteTech/releases/tag/RemoteTech-1.8.2

**Comment by NathanKell** [comment]
Checked the ones that are either released or (RF) ready to be.

@jwvanderbeck as Strato says, sorry, dunno yet. :\

**Comment by lamont-granquist** [comment]
RT released 1.8.3 now which is 1.2.2 compatible:  https://github.com/RemoteTechnologiesGroup/RemoteTech/releases/tag/1.8.3

**Comment by stratochief66** [comment]
@lamont-granquist thanks for checking RT. I personally don't play with it, so I wouldn't know how to test it. Have you personally used it in 1.2.2, ideally with RSS?

**Comment by lamont-granquist** [comment]
@stratochief66 no, i'm not playing RSS in 1.2.2, shaving other yaks...

**Comment by PhineasFreak** [comment]
RT works, if i can say, better in KSP 1.2 than in KSP 1.1.3. There are some major bug fixes (like the science transmittion bug) that have been fixed **and** they were backported to the 1.1.3 version.

From the list:

* RealHeat compiles and runs without any hitch or change from the previous release.
* Same with AJE (although there might be some optimizations to do).
* CBK has a 1.2 compatible release.
* PF has been updated with a lot of goodies and bug fixes.
* TestFlight has been updated by @linuxgurugamer but i have not tested it yet.

The rest of the mods need a rework.

**Comment by RedAV8R** [comment]
I know it's pretty minor, but PParts really could be updated to get that annoying popup to not popup. There will be people who continually say 'it's not updated', when while it works the popup shows says it's not. I've submitted a PR for it, though not entirely sure that's all that needs to happen.

**Comment by blowfishpro** [comment]
RF and AJE can be checked off now

**Comment by RedAV8R** [comment]
So RealHeat is checked off...is this now done internally with KSP or is there an actual updated version floating around somewhere?

**Comment by rsparkyc** [comment]
pushed up a PR for TACLS, anything else that I can help with?

**Comment by jwvanderbeck** [comment]
For TestFlight: https://github.com/KSP-RO/TestFlight/issues/154

**Comment by jwvanderbeck** [comment]
https://github.com/KSP-RO/TestFlight/releases/tag/1.8.0.0.beta-1

Needs to be tested

**Comment by rsparkyc** [comment]
Looks like there's a pre-release for DRE: https://github.com/Starwaster/DeadlyReentry/releases/tag/v7.5.0

**Comment by PhineasFreak** [comment]
**Note for future reference:** there have been some changes to SmokeScreen to add a new particle system (shuriken). Any RealPlume configs that do some patching using a "final" pass (i.e. zzRealPlume) will need to be updated:

**Before:**

```
@EFFECTS
{
    @Hydrolox-Lower
    {
        @MODEL_MULTI_PARTICLE_PERSIST[shockcone]
        {
            @localPosition = 0.0, 0.0, 0.0
        }
    }
}
```

**After:**

```
@EFFECTS
{
    @Hydrolox-Lower
    {
        @MODEL_MULTI_SHURIKEN_PERSIST[shockcone]
        {
            @localPosition = 0.0, 0.0, 0.0
        }
    }
}
```

**Comment by stratochief66** [comment]
@PhineasFreak do you know if there was a functional reason to change the node name? Will plumes require tweaking in size, shape, speed etc along with the new particle effect? If so, that makes sense to distinguish between old and new configs.

Is there any reason to not just run a script searching for _PARICLE_ in plumes and changing it to _SHURIKEN_ ?

If that would work, @rsparkyc is good with scripts :)

**Comment by PhineasFreak** [comment]
@stratochief66 i think that **sarbian** intends the new system to be a drop-in replacement without any required config changes.

The script solution would be ideal. I think that **Nhawks** also did it that way for his RealPlume PR (or a Find/Replace in Notepad++).

**Comment by blowfishpro** [comment]
More info on `MODEL_MULTI_SHURIKEN_PERSIST` -
http://forum.kerbalspaceprogram.com/index.php?/topic/64987-121-smokescreen-270-extended-fx-plugin-dec-18th/&do=findComment&comment=2892140

It is intended to be a drop-in replacement, the main benefit being less memory allocation while playing (i.e. gc alloc).  Additional optimization can be done by reducing `particleCountLimit` (which is a new field) but that is not necessary.

I have implemented this change in B9, and everything seems to be working as it did before.  I think it's safe to implement in RF's RealPlume configs as well.

**Comment by rsparkyc** [comment]
Cool, I can run through those later and make the change.

**Comment by diego-treitos** [comment]
CLS released a version that claims to support 1.2: https://github.com/codepoetpbowden/ConnectedLivingSpace/releases/tag/1.2.4.0
Should it be marked as "done"?

**Comment by stratochief66** [comment]
@diego-treitos Not until I or somebody else tests it in game (with an RO-ish modded install) thoroughly and reports back. :)

**Comment by rsparkyc** [comment]
Trying to test stuff, I see realheat is marked as done, but it looks like the latest version is still for 1.1.3 (https://github.com/KSP-RO/RealHeat/releases/tag/v4.3).  Is there a 1.2.2 recompile somewhere, I should I do it myself?

**Comment by PhineasFreak** [comment]
@rsparkyc i have pushed an update for RealHeat to the official repository (https://github.com/KSP-RO/RealHeat/pull/2). Seems like that a simple recompile does the job!

**Comment by rsparkyc** [comment]
Yeah, I'm an idiot, didn't think to look at open PRs

**Comment by rsparkyc** [comment]
for anyone interested, i'm going to track my testing progress in this spreadsheet: https://docs.google.com/spreadsheets/d/19_dhwSioZODCCtMFD0kZOtbmbg--ryoE0nzoWJL2cRc/edit?usp=sharing Right now i'm just trynig to get the "Mandatory" mods, along with RSS, and see what works

**Comment by SirKeplan** [comment]
Seems like Procedural wings https://github.com/Crzyrndm/B9-PWings-Fork needs it's KSP version bumped to 1.2.2.

**Comment by PhineasFreak** [comment]
@rsparkyc MechJeb and RemoteTech have stable KSP 1.2.2 releases so you can also tag them as with "No Obvious Issues".

**Comment by rsparkyc** [comment]
Yeah, I'm just slowly setting up my environment, adding mods little by little, and making sure nothing goes crazy in the process, so wasn't going to mark that on my spreadsheet until I actually got something to orbit with those mods installed (Right, now, I'm saying that if I can get to orbit, then there wasn't anything obvious)

**Comment by Derica6350** [comment]
I as well have been testing with a few of the mods, i haven't found anything yet and it is stable with mechjeb and Remotetech. Is there a an issue with the mods necessary for RP-0 listed?

**Comment by PhineasFreak** [comment]
@rsparkyc BTW, did you find any problems with the recompiled RealHeat assembly? Just asking for feedback.

**Comment by rsparkyc** [comment]
Didn't see anything that was obviously wrong, but I don't really have any good test cases to see if it's working properly.  I did have the assembly loaded, and exploded during re-entry (with no heat shield), so I'm assuming things are working.

**Comment by stratochief66** [comment]
@Derica6350 not currently. I'll ask @NathanKell about that next time I see him. It would end up over on the RP-0 repo it one were made. :)
Here is the one from last major update. My memory is terrible, but KCT  (Kerbal Construction Time), CBK (Custom Barn Kit) and CTT (dunno offhand) were listed for that one.
https://github.com/KSP-RO/RP-0/issues/154

**Comment by rsparkyc** [comment]
Ok, so the biggest issue I see right now is that I get some errors when trying to have both TestFlight and Realism Overhaul loaded up in GameData.  Just Realism Overhaul seems to work (though I do get some errors about certain parts), but as soon as I try to add testflight, it breaks.  Perhaps we need to update some testflight configs in our parts? Not sure.

**Comment by stratochief66** [comment]
@jwvanderbeck, what specific logs that would be handy for diagnosing @rsparkyc 's observation?
@rsparkyc which version of TestFlight are you using? 1.8.0.0 Beta 1 ?

**Comment by rsparkyc** [comment]
Yep, here's a better error message:
```
AmbiguousMatchException: Ambiguous matching in method resolution
  at System.Reflection.Binder.FindMostDerivedMatch (System.Reflection.MethodBase[] match) [0x00000] in <filename unknown>:0
  at System.MonoType.GetMethodImpl (System.String name, BindingFlags bindingAttr, System.Reflection.Binder binder, CallingConventions callConvention, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) [0x00000] in <filename unknown>:0
  at System.Type.GetMethod (System.String name, BindingFlags bindingAttr) [0x00000] in <filename unknown>:0
  at RealFuels.TestFlightWrapper..cctor () [0x00000] in <filename unknown>:0
Rethrow as TypeInitializationException: An exception was thrown by the type initializer for RealFuels.TestFlightWrapper
  at RealFuels.Tanks.ModuleFuelTanks.UpdateTestFlight () [0x00000] in <filename unknown>:0
  at RealFuels.Tanks.ModuleFuelTanks.UpdateTankType (Boolean initializeAmounts) [0x00000] in <filename unknown>:0
  at RealFuels.Tanks.ModuleFuelTanks.GetInfo () [0x00000] in <filename unknown>:0
  at PartLoader.CompilePartInfo (.AvailablePart newPartInfo, .Part part) [0x00000] in <filename unknown>:0
  at PartLoader+<CompileParts>c__Iterator62.MoveNext () [0x00000] in <filename unknown>:0
  at UnityEngine.SetupCoroutine.InvokeMoveNext (IEnumerator enumerator, IntPtr returnValueAddress) [0x00000] in <filename unknown>:0
```

**Comment by jwvanderbeck** [comment]
The problem there is RealFuels. I believe there was a new build to fix it.

**Comment by rsparkyc** [comment]
Thanks, let me look for that!

**Comment by blowfishpro** [comment]
I have not yet released a fix for that bug, but the fix has been completed.  There are a few other minor issues I want to take a look at first

**Comment by PhineasFreak** [comment]
Yep, i logged the error and a test assembly was kindly compiled by @blowfishpro.

**Comment by rsparkyc** [comment]
@blowfishpro, thx for the reply, should I recompile off of this commit? https://github.com/NathanKell/ModularFuelSystem/commit/ae3a9ad6980e76e1f4b13d9b8302de3018dd3f2b

**Comment by blowfishpro** [comment]
Yes, that commit has the fix.  I also posted a compiled DLL with the fix in this issue: https://github.com/NathanKell/ModularFuelSystem/issues/169

**Comment by rsparkyc** [comment]
Are there plans to update Procedural Fairings for Everything? (https://github.com/KSP-RO/ProceduralFairings-ForEverything)

**Comment by rsparkyc** [comment]
With my current setup (see spreadsheet), something's causing a NRE when I enter and exit the VAB
```
NullReferenceException: Object reference not set to an instance of an object
  at KSPAssets.Loaders.AssetLoader+<LoaderRoutine>d__53.MoveNext () [0x00000] in <filename unknown>:0
  at UnityEngine.SetupCoroutine.InvokeMoveNext (IEnumerator enumerator, IntPtr returnValueAddress) [0x00000] in <filename unknown>:0
UnityEngine.MonoBehaviour:StartCoroutine_Auto(IEnumerator)
UnityEngine.MonoBehaviour:StartCoroutine(IEnumerator)
KSPAssets.Loaders.AssetLoader:LoadAssets(Action`1, AssetDefinition[])
TestFlightCore.TestFlightKSCWindow:Start()
```

**Comment by rsparkyc** [comment]
When choosing the `LR87-LH2 Vacuum`, FAR gets unhappy:

```
*U* Ullage constructor called on LR87LH2Vac

(Filename: /Users/builduser/buildslave/unity/build/artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 42)

NullReferenceException: Object reference not set to an instance of an object
  at FerramAerospaceResearch.FARPartGeometry.GeometryModification.StockJettisonTransformGeoUpdater..ctor (.ModuleJettison engineFairing, FerramAerospaceResearch.FARPartGeometry.GeometryPartModule geoModule) [0x00000] in <filename unknown>:0
  at FerramAerospaceResearch.FARPartGeometry.GeometryPartModule.SetupIGeometryUpdaters () [0x00000] in <filename unknown>:0
  at FerramAerospaceResearch.FARPartGeometry.GeometryPartModule.Start () [0x00000] in <filename unknown>:0

(Filename:  Line: -1)
```

**Comment by rsparkyc** [comment]
More initial testing:

Lighting on my fairings seems off...
http://imgur.com/ZWa2NO5
Also, wasn't able to decouple the fairings on my payload (I think that was mentioned before somewhere)

**Comment by PhineasFreak** [comment]
@rsparkyc do note that PF has two options for fairing sides, the normal payload fairings (stagable) and the interstage fairings (non stagable). The second category is used for making interstages and various other structures that **must not** decouple their sides.

**Comment by rsparkyc** [comment]
I'll have to see which fairings I used

**Comment by rsparkyc** [comment]
Ok, I seem to have a "working" realism overhaul installation, now on to RP-0 (I added a new tab to my spreadsheet)

**Comment by Derica6350** [comment]
I have a few different KSP versions: 1.2.0.1586, 1.2.1 and 1.2.2 availiable and was wondering which one this is being built for specifically.

**Comment by kiwivogel** [comment]
@Derica6350 presumably the latest

**Comment by Derica6350** [comment]
I haven't looked into the changes between 1.2 and 1.2.2, what version have you been running on? @rsparkyc

**Comment by SirKeplan** [comment]
It will be for the latest version of 1.2 @Derica6350

**Comment by rsparkyc** [comment]
All my testing was on 1.2.2

**Comment by rsparkyc** [comment]
Ok, so I installed a bunch of RP-0 mods, and now I'm noticing that I can't transmit science data (it transmits, but I don't get the science points for it).  Can anyone see if they're getting the same thing w/o the RP-0 mods installed?

**Comment by Patrykz94** [comment]
@rsparkyc This looks like Remote Tech issue although according to the change logs, it was already fixed in 1.8.3 (and 1.7.2 for KSP 1.1.3). I haven't tested it so can't confirm but here's the releases page for RT: https://github.com/RemoteTechnologiesGroup/RemoteTech/releases

**Comment by rsparkyc** [comment]
@Patrykz94 yeah, I thought the same thing too.  Have an issue here for now: https://github.com/KSP-RO/RP-0/issues/607
Will probably create a stock install, load up RemoteTech, and see if i can duplicate w/o any other mods.

**Comment by blowfishpro** [comment]
Just released a fix for a few RF issues.  If you're testing be sure to get it.

**Comment by raidernick** [comment]
We need to add fixing the modulercs for all parts to this list. Stock now uses ModuleRCSFX on all it's parts. Old mods that aren't updated can keep the old modules but all mods that changed over need to have their configs updated to change the module names or the parts will be broken.

**Comment by PhineasFreak** [comment]
@raidernick can't we just replace all the MM checks of ModuleRCS with ModuleRCS* ? Since the basic operation remains identical and we currently do not implement any custom plumes for them.

**Comment by raidernick** [comment]
@PhineasFreak the thing is rcsfx DOES use a different plume. It has it's own fx file that needs to be defined. I did this in my own mod files, but just changing the name won't do anything. It would either have no plume, or you'd see the new fx plume over the old particles.

**Comment by ec429** [comment]
Is KSCSwitcher a listed RO dependency?  It probably needs https://github.com/KSP-RO/KSCSwitcher/pull/2 and a release.

**Comment by jwvanderbeck** [comment]
TestFlight is not longer pre-release
https://github.com/KSP-RO/TestFlight/releases/tag/1.8.0.0

**Comment by PhineasFreak** [comment]
Pinging the members for the RSSTimeDateFormatter. A removal of the "KSPUtil" reference and a recompile is all that is needed. I can also open a PR if you want.

**Comment by jwvanderbeck** [comment]
> Pinging the members for the RSSTimeDateFormatter. A removal of the "KSPUtil" reference and a recompile is all that is needed. I can also open a PR if you want.

That is mine.  If you can submit a PR to fix it that would be great otherwise I will look at it this weekend.

**Comment by PhineasFreak** [comment]
@jwvanderbeck Yep, i just forked the repo and i saw that the only thing required was a recompile. I had recompiled it locally when KSP 1.2.0 was originally released but i had forgotten that it did not require any changes to the code.

**Comment by jwvanderbeck** [comment]
Having some problems with Xamarin but will get it recompiled soon as I can.

**Comment by eberkain** [comment]
Consider [MandatoryRCS](http://forum.kerbalspaceprogram.com/index.php?/topic/154658-122-mandatoryrcs-12-1801-rw-rebalance-sas-rotation-persistence/) mod in place of Persistant Rotation?

**Comment by Theysen** [comment]
Why? Dare to elaborate eberkain? Is Persistent Rotation out of development and not working in 1.2.2?

**Comment by eberkain** [comment]
No, Persistent Rotation is fine.  I only suggested looking at what both mods offer and considering which one would be better to include.   Didn't mean to make anyone mad.

**Comment by Theysen** [comment]
Who got mad? I just wanted to know if you knew more than I did and might enlighten me, that was without any mad connotation!

**Comment by RedAV8R** [comment]
Considering that RO basically nukes Reaction Wheels and makes it so that RCS is essentially required I see no reason to include Mandatory RCS.

**Comment by RedAV8R** [comment]
@jwvanderbeck @PhineasFreak Any luck on RSS Time Date Formatter?

**Comment by PhineasFreak** [comment]
@RedAV8R I think that @jwvanderbeck will handle the required recompile. I can also open a PR to do so if he is busy.

**Comment by jwvanderbeck** [comment]
The problem I had was that I couldn't get it to build.  My Xamarin project was being stupid and kept insisting that .NET was missing and I couldn't fix it :(  I spent like a day trying to figure out what was wrong (TF builds fine) but couldn't figure it out.

I'll try just wiping what I have and download it fresh from git and see if that works.

**Comment by PhineasFreak** [comment]
@jwvanderbeck No probs. I also remember that one time that Xamarin acted stupidly after an update and would refuse to compile any of the projects that i had. Proper head banging stuff!

It turned out that i had to uninstall and reinstall it in order for the required Microsoft Build Tools to also install correctly...

**Comment by stratochief66** [comment]
Possibly unimportant, but I found that the previous version simply seems to work for me in KSP 1.2.2 ; maybe just needs a CKAN metadata change rather than recompile?

**Comment by jwvanderbeck** [comment]
https://github.com/KSP-RO/RSSTimeFormatter/releases/tag/1.1.0.0

**Comment by lamont-granquist** [comment]
RealHeat, ConnectedLivingSpaces and Deadly Reentry are all installable on 1.2.2 off of CKAN

**Comment by MuricanJeb** [comment]
The Ferram Aerospace Research (FAR) mod is updated!

**Comment by stratochief66** [comment]
@NathanKell @pap1723 @rsparkyc @SirKeplan 

So, all the pre-reqs for RO now appear to be available on CKAN now. Am I wrong?

Is RO ready to go for a full release? If no objects, I am thinking of making a release either tonight or tomorrow. Thoughts?

**Comment by pap1723** [comment]
Both RO and RP-0?

**Comment by NathanKell** [comment]
RO yay! But I would say RO only?

**Comment by Theysen** [comment]
KCT is still in beta develop so that rules out RP-0 for official CKAN / release

**Comment by Olympic1** [comment]
> KCT is still in beta develop so that rules out RP-0 for official CKAN / release

KCT is listed as a recommendation.

**Comment by PhineasFreak** [comment]
@Olympic1 this should probably be changed to a dependency, as RP-0 pretty much requires it for a balanced career gameplay.

**Comment by Theysen** [comment]
oh and RE: CLS - I didn't run into any issues with the latest 1.2.2 release.

**Comment by Olympic1** [comment]
KCT just released for KSP 1.2. That makes all dependencies available for RP-0

**Comment by NathanKell** [comment]
Released
